mlir/lib/Dialect/Rock/Transforms/LayoutEmitter.cpp

```diff
@@ -0,0 +1,80 @@
+//===- AccelEmitter.cpp - MLIR helper to emit acceleration intrinsics
+//---------------===//
+//
+// Copyright 2020 The MLIR Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =============================================================================
+//
+// This class tries to abstract away the code-generation details needed to
+// generated calls to matrix multiply accelerator intrinsics (wmma, mfma).
```

Comment:
As this is not a transform pass, IMO this file should live in utility? 
(same could be said for AccelEmitter)

Reply:
I think Transforms/ isn't required to just be passes, it can also have pass-specific support code

Reply:
which I always thought why Transforms/Utils/ or Utils/ exist... (but yea it is not strictly followed upstream either)
Anyways I dont have a strong opinion; just felt nice to have an criterion to decide which goes where.

Reply:
Yeah I don't know, I mean my (also not strong) preference is more toward a flat structure. I guess if we reach a point where we have a lot of utility code, then we can move everything in a separate folder